test/suite/test_checkpoint_snapshot05.py

```diff
@@ -140,38 +141,43 @@ def test_checkpoint_snapshot(self):
             else:
                 cursor1.set_value(self.valueb + str(i))
             self.assertEqual(cursor1.update(), 0)
-
-        # Create a checkpoint thread
+        
+        # Commit the transaction during a checkpoint.
         done = threading.Event()
         ckpt = checkpoint_thread(self.conn, done, checkpoint_count_max=1)
         try:
             ckpt.start()
             
-            # Wait for checkpoint to start and acquire its snapshot before committing.
+            # Wait for checkpoint to acquire its snapshot executing the commit.
             ckpt_snapshot = 0
             while not ckpt_snapshot:
-                time.sleep(1)
-                stat_cursor = self.session.open_cursor('statistics:', None, None)
-                ckpt_snapshot = stat_cursor[stat.conn.checkpoint_snapshot_acquired][2]
-                stat_cursor.close()
+                with open_cursor(self.session, 'statistics:') as stat_cursor:
+                    ckpt_snapshot = stat_cursor[stat.conn.checkpoint_snapshot_acquired][2]
+                time.sleep(0.5)
```

Comment:
Why don't we reduce the time even more?

Reply:
Certainly, but reduction by how much is enough? With `timing_stress_for_test=[checkpoint_slow]` it seems like the reduction 1->0.5 is really cosmetic. I did not have time to verify for how long `checkpoint_snapshot_acquired` signal is valid. If it is from the time it is acquired until checkpoint completion (which is what I am guessing) that is probably the best window we can have. Not infallible but reliable enough.

Reply:
Since we are heavily dependent on this event, I would just reduce it to as close as possible to 0. We have tests doing `sleep(0.1)`, I guess we could do this here too!